Dev.remove gsutil

saltyseadawg opened the pull request:
refactor to download packages with google-cloud-storage instead of gsutil
- add tests
- add black to lint command
- user must pass in text file with requirements instead of specifying package name

saltyseadawg committed:
refactor gs_pip_install to not use gsutil

saltyseadawg committed:
reorganize requirements files and fix arg ordering

saltyseadawg committed:
update requirements in setup

saltyseadawg committed:
remove unnecessary req file

saltyseadawg committed:
lint code and bug fixes
- add black to dev requirements
- pin click 7.1.2 for compatibility with black
- install dependencies if target directory isn't specified

saltyseadawg committed:
add docstring and increase max line length to 85

saltyseadawg committed:
add tests and elaborate help description for cli

saltyseadawg committed:
update readme with sample cli calls

lucidsushi commented on README.rst:
```diff
@@ -26,11 +26,10 @@ Pip Install Packages Stored in Google Cloud Buckets
-        gs_pip_install --bucket_url gs://python-package-location --package_name google_cloud_utils==1.0.1
+        gs_pip_install --project_id gcp_project --bucket_name python-package-location 
+        gs_pip_install --project_id gcp_project --bucket_name python-package-location --target_dir package_install_destination
+        gs_pip_install --project_id gcp_project --bucket_name python-package-location --req_file requirements_custom.txt
```
match pip signature

https://pip.pypa.io/en/stable/reference/pip_install/#install-requirement

lucidsushi commented on README.rst:
```diff
@@ -26,11 +26,10 @@ Pip Install Packages Stored in Google Cloud Buckets
 .. code-block:: bash
 
-        gs_pip_install --bucket_url gs://python-package-location --package_name google_cloud_utils==1.0.1
+        gs_pip_install --project_id gcp_project --bucket_name python-package-location 
```
don't force requirements.txt requirement

lucidsushi commented on gs_pip_install/gs_pip_install.py:
```diff
@@ -1,92 +1,116 @@
+    Args:
+        packages_download_dir (str): Local directory to download packages into
+        bucket_name (str): Name of GCS bucket to download packages from
+        project_id (str): Name of GCP project
```
make sure if we actually need this -- hopefully not

lucidsushi commented on Makefile:
```diff
@@ -47,7 +47,8 @@ clean-test: ## remove test and coverage artifacts
 
-lint: ## check style with flake8
+lint: 
+	black -vS gs_pip_install tests
```
add black config toml to match flake8 length

lucidsushi commented on gs_pip_install/gs_pip_install.py:
```diff
@@ -1,92 +1,116 @@
-    ):
-        subprocess.call(cmd)
+        install_packages(download_dir, target_dir)
+    finally:
```
@saltyseadawg  nice  try/finally

lucidsushi commented on README.rst:
```diff
@@ -26,11 +26,10 @@ Pip Install Packages Stored in Google Cloud Buckets
 
```
@saltyseadawg  in the future we can quickly review call signature like this first (pilot documentation), save some pain for fixing up tests at review phase

saltyseadawg committed:
refactor cli arguments and downloading of packages
- add short and explicit options to cli
- enable user to install a package by its name or by passing in a text file listing packages
- refactor tests

saltyseadawg committed:
add config file for black
